src/python/pants/backend/codegen/protobuf/python/grpc_python_plugin.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pants.core.util_rules.external_tool import ExternalTool
+from pants.engine.platform import Platform
+from pants.util.enums import match
+
+
+class GrpcPythonPlugin(ExternalTool):
+    """The grpc protobuf plugin."""
+
+    options_scope = "grpc_python_plugin"
+    default_version = "1.32.0"
+    default_known_versions = [
+        "1.32.0|darwin|b2db586656463841aa2fd4aab34fb6bd3ef887b522d80e4f2f292146c357f533|6215304",
+        "1.32.0|linux |1af99df9bf733c17a75cbe379f3f9d9ff1627d8a8035ea057c3c78575afe1687|4965728",
+    ]
+
```

Comment:
You should also implement `generate_exe`. We stopped using a default implementation under the reasoning that it's more clear to be explicit at each subclass.

Reply:
This just downloads the file you need. `generate_exe` is for when the download is an archive, and you need to locate the binary within that archive.

Reply:
Do we expect the user to call `downloaded_tool.exe`? If so, we need to implement `generate_exe`, its `@abstractmethod` otherwise.

E.g. Pex:

https://github.com/pantsbuild/pants/blob/d526d4e5e9ae6887cdd038b208cb0c40fc64207f/src/python/pants/backend/python/util_rules/pex_cli.py#L54-L55

Reply:
Hmm, I think the semantics have changed since I wrote ExternalTool. In principle if you are downloading the file itself, not an archive, then you don't need `exe`, but it looks like now you have to implement it. So the docstring on generate_exe is wrong.

Reply:
Looking at the history here, I think the default was fine, why was it changed? If downloading an executable file directly, not in an archive, we already know its name, so why force people to add boilerplate?

Reply:
To be explicit that that's what's going on. I found it a bit magical before, that you had to look at the implementation of `ExternalTool` to be confident `exe` was right. Stu acked the change. 

Feel free to change it back, this probably isn't a big deal either way.

Reply:
I mean, it's always right if the download is the executable itself. Maybe the confusion is that the old code was on the caller side, and really this should have been a default implementation of `generate_exe()`. I think that would make it more obvious.